addons/website_sale/static/src/js/website_sale_utils.js

```diff
@@ -32,7 +36,47 @@ function animateClone($cart, $elem, offsetTop, offsetLeft) {
     }
 }
 
+/**
+ * Allows to propose the existing products from the database.
+ *
+ * @param {ServicesMixin|Widget} self - an element capable to trigger an RPC
+ * @param {jQuery} $input
+ */
+function autocompleteWithProducts(self, $input, data) {
+    $input.autocomplete({
+        source: function (request, response) {
+            return self._rpc({
+                route: '/website_sale/get_products_info',
+                params: {
+                    term: request.term,
+                    limit: 7,
+                    product_options: {
+                        'display_description': data.displayDescription,
+                        'display_price': data.displayPrice,
+                        'display_image': data.displayImage,
+                        'order_by': data.orderBy,
+                    }
+                },
+            }).then((function (resources) {
+                this.products_options = resources.products_options;
```

Comment:
I would not use `this`, it is difficult to know what it is (I am still not sure). Declare the variables in the `autocompleteWithProducts` function and set them here.